### PR TITLE
replace custom logger with env_logger

### PR DESCRIPTION
**Pull request by michaelkirk**
It's nice to be able to adjust log levels without re-compiling/linking.

    RUST_LOG=debug cargo run --bin game

Note that this can be done on a per lib basis:

    RUST_LOG=my_lib=debug cargo run --bin game

a module-by-module basis:

    RUST_LOG=my_lib::module=debug cargo run --bin game

And you can mix and match:

    # error logging by default, except the foo:bar module at debug level
    # and the entire baz crate at info level
    RUST_LOG=error,foo::bar=debug,baz=info cargo run --bin game

This is the logging flow that I like - I often spend some significant
time implementing fairly verbose and structured logging while developing
a feature, only to delete it so as not to spam us in general. With this
approach I can leave more of the useful logging in and enable debug
level for just the modules I'm focused on.

Note that this loses the special-case throttling behavior for fastpaths
logging. Now that default level is info we're not seeing any of this output
anyway. But if we want it - an alternative to implementing our custom logger
could use env_loggers regex handling:

    # only log once every 10k
    RUST_LOG="fast_paths=debug/contracted node [0-9]+0000 " mike import_la

It's a bit unpleasant, but seems like it could work and uses standard stuff.

```
[2020-12-07T19:11:10Z DEBUG fast_paths::fast_graph_builder] contracted node 10000 / 31505, num edges fwd: 18009, num edges bwd: 19719
[2020-12-07T19:11:10Z DEBUG fast_paths::fast_graph_builder] contracted node 20000 / 31505, num edges fwd: 19381, num edges bwd: 21131
[2020-12-07T19:11:13Z DEBUG fast_paths::fast_graph_builder] contracted node 30000 / 31505, num edges fwd: 48427, num edges bwd: 50510                  
prepare pathfinding for bikes took 8.8064s                                                                                                             
prepare pathfinding for buses...
[2020-12-07T19:11:19Z DEBUG fast_paths::fast_graph_builder] contracted node 10000 / 31505, num edges fwd: 20422, num edges bwd: 21676
[2020-12-07T19:11:19Z DEBUG fast_paths::fast_graph_builder] contracted node 20000 / 31505, num edges fwd: 21710, num edges bwd: 22990
[2020-12-07T19:11:19Z DEBUG fast_paths::fast_graph_builder] contracted node 30000 / 31505, num edges fwd: 49486, num edges bwd: 50139                  
prepare pathfinding for buses took 0.7516s                                                                                                            
prepare pathfinding for trains...
[2020-12-07T19:11:19Z DEBUG fast_paths::fast_graph_builder] contracted node 10000 / 31505, num edges fwd: 0, num edges bwd: 2
```
Read more: https://docs.rs/env_logger/0.8.2/env_logger/

### CODE REVIEW

File: abstutil/Cargo.toml [hunk 1]
```diff
@@ -10,6 +10,7 @@ instant = "0.1.7"
 itertools = "0.9.0"
 lazy_static = "1.4.0"
 log = { version = "0.4.11", features=["std"] }
+env_logger = { version = "0.8.2" }
```
Comment (dabreegster) [inline abstutil/Cargo.toml:13]: nit: Please sort

File: abstutil/src/logger.rs [hunk 2]
```diff
@@ -1,74 +1,16 @@
...
-
-        #[cfg(not(target_arch = "wasm32"))]
-        {
-            log::set_boxed_logger(Box::new(Logger {
-                last_fast_paths_note: RwLock::new(None),
-            }))
-            .unwrap();
-            log::set_max_level(log::LevelFilter::Info);
-        }
+/// On native: intercept messages using the `log` crate and print them to STDOUT. Contains
```
Comment (dabreegster) [inline abstutil/src/logger.rs:1]: Update docstring to something like: Initialize `env_logger` on native and `console_log` on web.

**Review by dabreegster — COMMENTED** [review]
Very nice! Could we also update `book/src/dev/README.md` in the `Development tips` section to mention that we use env_logger? Some of the examples in your PR description could be nice there for reference, or at a minimum, just mention running with `RUST_LOG=debug`. Thanks!

> Note that this loses the special-case throttling behavior for fastpaths logging.

I think it's worth adding a note to `book/src/dev/README.md` in the `Building map data` section, along the lines of: "Building contraction hierarchies for pathfinding occurs in the `--map` stage. It can take a few minutes for larger maps. To view occasional progress updates, you can run the importer with `RUST_LOG="fast_paths=debug/contracted node [0-9]+0000 "`.

File: abstutil/Cargo.toml (hunk 1 above)
Comment (michaelkirk) [reply abstutil/Cargo.toml]: Sorry that I always forget to do this. Fixed!

File: abstutil/src/logger.rs (hunk 2 above)
Comment (michaelkirk) [reply abstutil/src/logger.rs]: Done - plus added some examples. It's mostly duplicated w/ what I added to the dev docs, but I don't know where people are more likely to look. Happy to remove one if you'd prefer to keep it DRY.

**Comment by michaelkirk** [comment]
OK - this is ready for re-review. PTAL @dabreegster

**Comment by dabreegster** [comment]
Awesome, thanks for fleshing out the documentation!

**Pull request merged**